compose/Dockerfile.source

```diff
@@ -0,0 +1,266 @@
+RUN tar --extract --gzip --file=/usr/local/src/cardano-cli/cardano-cli-${CARDANO_CLI_VERSION}-x86_64-linux.tar.gz --directory=/usr/local/src/cardano-cli
+
+# Make cardano-cli executable
+RUN chmod 0755 /usr/local/src/cardano-cli/cardano-cli-x86_64-linux
+
+# Create cardano-cli symlink
+RUN ln -s /usr/local/src/cardano-cli/cardano-cli-x86_64-linux /usr/local/bin/cardano-cli
+
+# Create uv source directory
+RUN install --directory --owner=root --group=root --mode=0755 /usr/local/src/uv
+
+# Download uv archive
+RUN curl --proto '=https' --tlsv1.2 \
+        --location https://github.com/astral-sh/uv/releases/download/${UV_VERSION}/uv-x86_64-unknown-linux-gnu.tar.gz \
+        --output /usr/local/src/uv/uv-x86_64-unknown-linux-gnu.tar.gz
+
+# Download uv checksum
+RUN curl --proto '=https' --tlsv1.2 \
+        --location https://github.com/astral-sh/uv/releases/download/${UV_VERSION}/uv-x86_64-unknown-linux-gnu.tar.gz.sha256 \
+        --output /usr/local/src/uv/uv-x86_64-unknown-linux-gnu.tar.gz.sha256
+
+# Verify uv checksum
+WORKDIR /usr/local/src/uv
+RUN sha256sum --ignore-missing --check uv-x86_64-unknown-linux-gnu.tar.gz.sha256
+
+# Unarchive uv tarball
+RUN tar --extract --gzip --file=/usr/local/src/uv/uv-x86_64-unknown-linux-gnu.tar.gz --directory=/usr/local/src/uv
+
+# Make uv executable
+RUN chmod 0755 /usr/local/src/uv/uv-x86_64-unknown-linux-gnu/uv
+
+# Create uv symlink
+RUN ln -s /usr/local/src/uv/uv-x86_64-unknown-linux-gnu/uv /usr/local/bin/uv
+
+# Clone testnet-generation-tool.git repository
+WORKDIR /usr/local/src
+RUN git clone --branch main https://github.com/cardano-foundation/testnet-generation-tool.git
+
+# Download testnet-generation-tool dependencies
+WORKDIR /usr/local/src/testnet-generation-tool
+RUN uv sync
+
+# Copy testnet.yaml specification
+COPY --chown=root:root .testnet.yaml /usr/local/src/testnet-generation-tool/testnet.yaml
+
+# Build testnet configuration files
+RUN uv run python3 genesis-cli.py testnet.yaml -o /tmp/testnet -c generate
+
+# Remove dynamic topology.json
+RUN find /tmp/testnet -type f -name 'topology.json' -exec rm -f '{}' ';'
+
+#---------------------------------------------------------------------
+
+FROM docker.io/debian:stable-slim AS main
+
+# Set environment variables
+ENV CARDANO_NODE_SOCKET_PATH="/opt/cardano-node/data/db/node.socket"
+
+# Set time zone
+ENV TZ="UTC"
+RUN ln -snf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+    echo ${TZ} > /etc/timezone
+
+# Install packages required in main stage
+RUN apt update && \
+    apt install -y --no-install-recommends \
+        dnsutils \
+        iproute2 \
+        iputils-ping \
+        jq \
+        less \
+        lsof \
+        netbase \
+        netcat-openbsd \
+        openssl \
+        procps \
+        tcpdump \
+        telnet \
+        vim
+
+# Copy libsodium library
+COPY --from=build --chown=root:root /usr/local/lib/libsodium.so /usr/local/lib
+RUN ln -snf /usr/local/lib/libsodium.so /usr/local/lib/libsodium.so.23 && \
```

Comment:
Minor: Compiling Cardano-node against static versions of the crypto libs would alleviate the need to copy them here

Reply:
Linking against static libs in haskell is normally a nightmare. This version works.

Reply:
Building a fully statically linked Haskell program is indeed a nightmare, linking to static libraries is actually trivial, I did it [here](https://github.com/pragma-org/db-server/blob/main/build.hs#L95) for a Haskell program using consensus code and therefore depending on the holy trinity (lib sodium, libsecp256k, and libblst).
This was just a suggestion that could simplify things a bit but of course it works with dynamic libs too